File: app/src/main/java/com/androidauthority/a2dgame/Villain.java
package com.androidauthority.a2dgame;

import android.content.Context;
import android.content.res.Resources;
import android.graphics.Bitmap;
import android.graphics.BitmapFactory;
import android.graphics.Canvas;
import android.graphics.Color;
import android.graphics.Paint;
import android.graphics.Rect;
import android.util.DisplayMetrics;
import android.util.Log;

import java.util.Random;

public class Villain {

    private final Context context;
    private final Bitmap image;
    public Rect rect;
    private int screenWidth;
    private int screenHeight;
    private int velocity;


    public Villain(Context context) {
        this.context = context;
        this.image = BitmapFactory.decodeResource(this.context.getResources(), R.drawable.enemy);
        int imageHeight = this.image.getHeight();
        int imageWidth = this.image.getWidth();
        Random r = new Random();
        screenHeight = Resources.getSystem().getDisplayMetrics().heightPixels;
        screenWidth = Resources.getSystem().getDisplayMetrics().widthPixels;
        velocity = r.nextInt(10) + 3;
        int bounds = 50;
        int left = r.nextInt(screenWidth - imageWidth - 2*bounds) + bounds;
        int right = left + imageWidth;
        int bottom = 0;
        int top = bottom - imageHeight;
        this.rect = new Rect(left, top, right, bottom);
    }

    public void draw(Canvas canvas) {
        // canvas.drawBitmap(image, rect.left, rect.top, null);
        Paint p = new Paint();
        p.setStyle(Paint.Style.FILL_AND_STROKE);
        p.setColor(Color.RED);
        canvas.drawRect(this.rect, p);
    }

    public void update() {
        int top = rect.top + velocity;
        int bottom = rect.bottom + velocity;
        rect.set(rect.left, top, rect.right, bottom);
    }
}

File: app/src/main/java/com/androidauthority/a2dgame/Villain.kt
package com.androidauthority.a2dgame

import android.content.Context
import android.content.res.Resources
import android.graphics.*
import java.util.*

class Villain(private val context: Context) {
    private val image: Bitmap = BitmapFactory.decodeResource(this.context.resources, R.drawable.enemy)
    var rect: Rect
    private val screenWidth: Int
    private val screenHeight: Int
    private val velocity: Int


    init {
        val imageHeight = this.image.height
        val imageWidth = this.image.width
        val r = Random()
        screenHeight = Resources.getSystem().displayMetrics.heightPixels
        screenWidth = Resources.getSystem().displayMetrics.widthPixels
        velocity = r.nextInt(10) + 3
        val bounds = 50
        val left = r.nextInt(screenWidth - imageWidth - 2 * bounds) + bounds
        val right = left + imageWidth
        val bottom = 0
        val top = bottom - imageHeight
        this.rect = Rect(left, top, right, bottom)
    }

    fun draw(canvas: Canvas) {
        // canvas.drawBitmap(image, rect.left, rect.top, null);
        val p = Paint()
        p.style = Paint.Style.FILL_AND_STROKE
        p.color = Color.RED
        canvas.drawRect(this.rect, p)
    }

    fun update() {
        val top = rect.top + velocity
        val bottom = rect.bottom + velocity
        rect.set(rect.left, top, rect.right, bottom)
    }
}


